### ISSUE TITLE
CTF: reset both flags if one is captured

### ISSUE THREAD

**Issue by ghost** [deferred-impact/battledudes-issue-tracker#234]
As I was playing a game, I had the enemy flag, and I noticed that our captures were tied with the enemy team. The enemy also had our flag, which meant that both teams were going to score at the same time, promoting a tie. Thus, I thought of an idea during that situation, that, what if, in a scenario where both flags were taken from both teams, the one to capture first, would force its own flag back to its spawn, as well as score a capture. 
So if the enemy stole our flag, and I stole the enemy's flag, and I somehow managed to capture before them, they would lose our flag, and also be behind by a capture. 

It would be an interesting idea, seeing that it would allow teams to be able to play defensively, and offensively at the same time, by being able to protect their flag, while also capturing the opponent's flag. 

Now I understand, this isn't an idea of utmost importance, it would be best if the developer was more focused on adding newer guns to the game, and improving the game, but I wished to add this, as another update that could be considered in the future.

**Comment by Haox09** [comment]
That would make people to defend even less and go full in offense (unsuccessfully)

**Comment by ghost** [comment]
Fair, because defense is a big part of the game, a specific part of it being which part of a team stays back for defense, and which part tries to capture. 

But at the same time, many people don't participate in defense anyways, so at least their offensive strategies and tries will allow them to be defensive at the same time. 
Plus the mechanic can't exactly be abused instantly, because **both flags need to be stolen in order for that to happen,** a team can't completely be offensive. 

But it has also come to my attention that if this were to be implemented, then most likely players would adapt their playing style to be completely offensive, by only worrying about capturing flags, and not defending at all, which means both teams would be at a rush to just capture flags, and it would be a game of which team captured first. 
So perhaps I will need to rethink my idea.

**Comment by deferred-impact** [comment]
I'm not particularly fond of this idea. I worked hard to get that flag from out of their base and all the way to my base. I don't want to have my work erased simply because i happened to do it at the wrong moment. 

This change would also discourage a slow and methodic approach to flag capture. Either you rush straight from enemy base to your base, or you lose the flag. That's idiotic. Players who are careful should be rewarded, not punished.

What i would rather see, is the condition that you can only capture the opposing flag if your own flag is safely chilling in the base. This would similarly encourage flag defense and discourage trades, but also wouldn't outright ban flag trades.

**Comment by ghost** [comment]
Yeah, sounds better, I think the flag condition to only be able to capture your flag is safe sounds better. 
However, there is one issue, enemies could continuously keep your flag away from spawn, to keep you from capturing, which is annoying. 
When an enemy steals your flag, and then dies, the flag stays at the spot for a while, and then returns to spawn. However, if enemies can continuously keep touching the flag, just to refresh the flag cooldown, it would prevent you from being able to capture as well. 
It also increases the risk of a flag holder not being able to capture, because then the enemy can just steal your flag, and then constantly attack you while also preventing you from capturing.

**Comment by deferred-impact** [comment]
> However, there is one issue, enemies could continuously keep your flag away from spawn, to keep you from capturing, which is annoying.

Could be a distance thing. Draw a circle around your flag spawn. If your flag is within that circle, you're allowed to capture. Once your flag leaves that circle, you cannot capture. Once your flag respawns, you again can capture.

> It also increases the risk of a flag holder not being able to capture, because then the enemy can just steal your flag, and then constantly attack you while also preventing you from capturing.

Which is... how the gamemode should be played anyway? What's the complaint exactly?

**Comment by ghost** [comment]
> Could be a distance thing. Draw a circle around your flag spawn. If your flag is within that circle, you're allowed to capture. Once your flag leaves that circle, you cannot capture it. Once your flag respawns, you again can capture.

And what if they take the flag outside the circle? They could easily carry it further, and then all they'd have to do, is keep refreshing the flag cooldown, until your team's flag holder dies. 
I was thinking that instead of not allowing players to capture the enemy flag when it isn't in its spawn, you would keep players from capturing only if their flag was possessed by the enemy team. Which means, if it is stationary, on cooldown in a location, then you may capture. 

> Which is... how the gamemode should be played anyway? What's the complaint exactly?

The problem is the same as the one you had, when I mentioned the idea of the faster capturer to get the point:

> I worked hard to get that flag from out of their base and all the way to my base. I don't want to have my work erased simply because i happened to do it at the wrong moment.

Why should I have to lose my hard earned work, simply because my own flag keeps getting touched, and is barely moving?

**Comment by deferred-impact** [comment]
> And what if they take the flag outside the circle? 

Then go fight them lmao

They can't capture either, because you are both carrying

This forces a fight between you two flag carriers

> I was thinking that instead of not allowing players to capture the enemy flag when it isn't in its spawn, you would keep players from capturing only if their flag was possessed by the enemy team. Which means, if it is stationary, on cooldown in a location, then you may capture.

That... swings the problem into the other direction, it's perfectly normal for a flag to be dropped mid capture, and with this, even if it's dropped for a second, suddenly your team captures. That's hardly fair.

> Why should I have to lose my hard earned work, simply because my own flag keeps getting touched, and is barely moving?

IDK, why did you open this issue to begin with? The answer to that would answer this as well

**Comment by ghost** [comment]
> Then go fight them lmao

> They can't capture either, because you are both carrying

> This forces a fight between you two flag carriers

Why would I go fight? That risks the flag being sent back. Simply forces both teams into a softlock, if they continue to keep the flag out of the circle, then neither will capture, and the game can't progress. 

> That's hardly fair.

how


> IDK, why did you open this issue to begin with? The answer to that would answer this as well

To allow players to be offensive and defensive. However I realized, it wasn't such a good idea, and thought of rethinking it. Then, you decided to add your random condition in exchange for my idea.

**Comment by deferred-impact** [comment]
> Why would I go fight?

Because if you don't, nothing happens, and the game is softlocked as you say

> Simply forces both teams into a softlock, 

No, if they actually go fight each other there will be no softlock

**Comment by ghost** [comment]
> Because if you don't, nothing happens, and the game is softlocked as you say

I'd rather be in a softlock and keep the flag, than risk the opponent's flag being sent back to its spawn.

**Comment by deferred-impact** [comment]
Be in a softlock then, lmao

**Comment by ghost** [comment]
ogey
